lib/step_flow/amqp/connection.ex

```diff
@@ -35,18 +38,58 @@ defmodule StepFlow.Amqp.Connection do
     {:noreply, conn}
   end
 
-  def handle_cast({:consume, queue, _callback}, conn) do
-    Logger.warn("#{__MODULE__}: consume messages on queue: #{queue}")
-    # AMQP.Queue.declare(conn.channel, queue, durable: false)
-    {:ok, _consumer_tag} = AMQP.Basic.consume(conn, queue)
-    {:noreply, conn}
-  end
+  # def handle_cast({:consume, queue, _callback}, conn) do
+  #   Logger.warn("#{__MODULE__}: consume messages on queue: #{queue}")
+  #   # AMQP.Queue.declare(conn.channel, queue, durable: false)
+  #   {:ok, _consumer_tag} = AMQP.Basic.consume(conn, queue)
+  #   {:noreply, conn}
+  # end
 
   def handle_info({:DOWN, _, :process, _pid, _reason}, _) do
     {:ok, chan} = rabbitmq_connect()
     {:noreply, chan}
   end
 
+  # Confirmation sent by the broker after registering this process as a consumer
+  def handle_info({:basic_consume_ok, %{consumer_tag: _consumer_tag}}, chan) do
+    {:noreply, chan}
+  end
+
+  def handle_info(
+        {:basic_deliver, payload, %{delivery_tag: tag, redelivered: _redelivered} = _headers},
+        channel
+      ) do
+    data =
+      payload
+      |> Jason.decode!()
+
+    job_id = Map.get(data, "job_id")
+
+    try do
+      Jobs.get_job(job_id)
+    rescue
+      _e in Ecto.NoResultsError ->
+        Logger.error("Cannot retrieve Job")
+
+      _e ->
+        AMQP.Basic.reject(channel.channel, tag, requeue: true)
+    end
```

Comment:
remove it, it do nothing interesting.
To get exception it requires to call `Jobs.get_job!(job_id)`